x/evm/types/msg.go

```diff
@@ -186,15 +240,20 @@ func (msg MsgEthereumTx) GetSignBytes() []byte {
 // The function will fail if the sender address is not defined for the msg or if
 // the sender is not registered on the keyring
 func (msg *MsgEthereumTx) Sign(ethSigner ethtypes.Signer, keyringSigner keyring.Signer) error {
+
 	from := msg.GetFrom()
 	if from.Empty() {
 		return fmt.Errorf("sender address not defined for message")
 	}
 
 	tx := msg.AsTransaction()
-	txHash := ethSigner.Hash(tx)
 
-	sig, _, err := keyringSigner.SignByAddress(from, txHash.Bytes())
+	chainID := ethSigner.ChainID()
+	msgbytes, computesignbyteserr := ComputeSignBytes(chainID, tx)
```

Comment:
I don't understand what's the reference for this implementation? The purpose of `msg.Sign` was to reproduce the `SignTx(tx *Transaction, s Signer, prv *ecdsa.PrivateKey) (*Transaction, error)` function from go-ethereum without the need of exposing the private key. This implementation changes the sign bytes as well

Reply:
it's exactly the same, with the previous one, just making sign bytes beforehand.
address is also the same

Reply:
what do you mean "without the need of exposing privatekey"?
it's using the same function "SignByAddress"?